monitoring.tf

```diff
@@ -79,20 +81,23 @@ resource "aws_lambda_permission" "with_sns" {
 }
 
 resource "aws_kms_key" "this" {
+  count                   = var.alarm_lambda_arn != "" ? 1 : 0
   description             = "KMS key is used to encrypt this sns topic"
   deletion_window_in_days = 7
   enable_key_rotation     = true
   policy                  = data.aws_iam_policy_document.kms.json
 }
 
 resource "aws_kms_alias" "backup_sns" {
+  count         = var.enable_sns_notifications ? 1 : 0
   name          = "alias/aws_backup-sns-${var.env}"
   target_key_id = aws_kms_key.this.arn
 }
 
 module "sns_topic" {
+  count   = var.enable_sns_notifications ? 1 : 0
```

Comment:
when you have count used you can access module/resource/data block outputs by using additional list ".[0]", so that following changes also should be there:
`data.aws_iam_policy_document.kms.json` => `data.aws_iam_policy_document.kms[0].json`
`aws_kms_key.this.arn` => `aws_kms_key.this[0].arn`
`module.sns_topic.sns_topic_arn` => `module.sns_topic[0].sns_topic_arn`
...

please do tests for this changes to make sure it works